core/unit_test/TestTeam.hpp

```diff
@@ -1553,12 +1556,67 @@ struct TestScratchAlignment {
         });
     Kokkos::fence();
   }
+
+  void test_minimal() {
+    using member_type = typename Kokkos::TeamPolicy<ExecSpace>::member_type;
+    Kokkos::TeamPolicy<ExecSpace> policy(1, 1);
+    size_t scratch_size = sizeof(int);
+    Kokkos::View<int, ExecSpace> flag("Flag");
+
+    Kokkos::parallel_for(
+        policy.set_scratch_size(0, Kokkos::PerTeam(scratch_size)),
+        KOKKOS_LAMBDA(const member_type &team) {
+          int *scratch_ptr = (int *)team.team_shmem().get_shmem(scratch_size);
+          if (scratch_ptr == nullptr) flag() = 1;
+        });
+    Kokkos::fence();
+    int minimal_scratch_allocation_failed = 0;
+    Kokkos::deep_copy(minimal_scratch_allocation_failed, flag);
+    ASSERT_TRUE(minimal_scratch_allocation_failed == 0);
+  }
+
+  void test_raw() {
+    using member_type = typename Kokkos::TeamPolicy<ExecSpace>::member_type;
+    Kokkos::TeamPolicy<ExecSpace> policy(1, 1);
+    Kokkos::View<int, ExecSpace> flag("Flag");
+
+    Kokkos::parallel_for(
+        policy.set_scratch_size(0, Kokkos::PerTeam(1024)),
+        KOKKOS_LAMBDA(const member_type &team) {
+          intptr_t scratch_ptr1 = intptr_t(team.team_shmem().get_shmem(24));
+          intptr_t scratch_ptr2 = intptr_t(team.team_shmem().get_shmem(32));
+          intptr_t scratch_ptr3 = intptr_t(team.team_shmem().get_shmem(12));
+
+          if (((scratch_ptr2 - scratch_ptr1) != 24) ||
+              ((scratch_ptr3 - scratch_ptr2) != 32))
+            flag() = 1;
+
+          if ((scratch_ptr3 + 12) % 8 == 4)
+            scratch_ptr1 = intptr_t(team.team_shmem().get_shmem_aligned(24, 4));
+          else {
+            scratch_ptr1 = intptr_t(team.team_shmem().get_shmem_aligned(12, 4));
+          }
+          scratch_ptr2 = intptr_t(team.team_shmem().get_shmem_aligned(32, 8));
+          scratch_ptr3 = intptr_t(team.team_shmem().get_shmem_aligned(8, 4));
+
+          if ((int(scratch_ptr2 - scratch_ptr1) != 28) &&
+              (int(scratch_ptr2 - scratch_ptr1) != 16))
+            flag() = 1;
+          if (int(scratch_ptr3 - scratch_ptr2) != 32) flag() = 1;
+          if ((int(scratch_ptr1 % 4) != 0) || (int(scratch_ptr2 % 8) != 0) ||
+              (int(scratch_ptr3 % 4) != 0))
```

Comment:
Do you need to cast to int here?  Same question above L1602-1603

Reply:
I propbably could write 4lu and 0lu instead ...